### ISSUE TITLE
Bad error reporting 

### ISSUE THREAD

**Issue by arichiardi** [grav/aws-lumo-cljs-runtime#1]
Hi there!

I was playing around with this project and I got the following errors in CloudWatch, kind of all of a sudden, after I saw it working a couple of times:

```
(node:12) UnhandledPromiseRejectionWarning: Error: Assert failed: Unexpected http://127.0.0.1:9001/2018-06-01/runtime/invocation/d9a9675a-0ac2-47ef-aab3-f142775739a7/errorresponse:
{
    "meta": null,
    "cnt": 3,
    "arr": [
        {
            "ns": null,
            "name": "body",
            "fqn": "body",
            "_hash": -2049205669,
            "cljs$lang$protocol_mask$partition0$": 2153775105,
            "cljs$lang$protocol_mask$partition1$": 4096
        },
        "",
        {
            "ns": null,
            "name": "status",
            "fqn": "status",
            "_hash": -1997798413,
            "cljs$lang$protocol_mask$partition0$": 2153775105,
            "cljs$lang$protocol_mask$partition1$": 4096
        },
        405,
        {
            "ns": null,
            "name": "headers",
            "fqn": "headers",
            "_hash": -835030129,
            "cljs$lang$protocol_mask$partition0$": 2153775105,
            "cljs$lang$protocol_mask$partition1$": 4096
        },
        {
            "meta": null,
            "cnt": 3,
            "arr": [
                "date",
                "Wed, 17 Apr 2019 00:18:17 GMT",
                "content-length",
                "0",
                "connection",
                "close"
            ],
            "__hash": null,
            "cljs$lang$protocol_mask$partition0$": 16647951,
            "cljs$lang$protocol_mask$partition1$": 139268
        }
    ],
    "__hash": null,
    "cljs$lang$protocol_mask$partition0$": 16647951,
    "cljs$lang$protocol_mask$partition1$": 139268
}
(= status 200)
at /tmp/tmp.mn6GxQr1qo:68:8
at process._tickCallback (internal/process/next_tick.js:68:7)
(node:12) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:12) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

I see there is a `status` `405` in there? What can be the cause?

This also made me realize that the trade-off of a custom layer is that you don't have visibility of these kind of things when they happen.

Nonetheless, thank you very much, this was very great to try out!

**Comment by arichiardi** [comment]
Ok I think I identified that the code either in `post-error` or `request` has something wrong because I receive an empty body in the response to the `(str runtime-path "/invocation/" aws-request-id "/error")` call

**Comment by arichiardi** [comment]
It is actually not entirely related to this project's code.

By exploring `rust`'s [implementation](https://github.com/awslabs/aws-lambda-rust-runtime/blob/master/lambda-runtime-client/src/client.rs#L220) I have found that:

> [...] the API returns 4xx errors for responses that are too long.

**Comment by arichiardi** [comment]
Great #3 solves all my issues

**Comment by grav** [comment]
Closed with #3
